renderdoc/data/glsl/mesh.vert

```diff
@@ -36,6 +36,69 @@
 #define SECONDARY_TYPE vec4
 #endif
 
+// This function is mostly duplicated between 'mesh.hlsl' and 'mesh.vert'.
+// Without a convenient shared common source, changes to one should be
+// reflected in the other.
+void vtxExploder(in int vtxID, inout vec3 pos, inout vec3 secondary)
+{
+  float nonLinearVtxExplodeScale = 4.0f * Mesh.exploderScale * Mesh.vtxExploderSNorm *
+                                   Mesh.vtxExploderSNorm * Mesh.vtxExploderSNorm;
+
+  // A vertex might be coincident with our 'exploderCentre' so that, when normalized,
+  // can give us INFs/NaNs that, even if multiplied by a zero 'exploderScale', can
+  // leave us with bad numbers (as seems to be the case with glsl/vulkan, but not hlsl).
+  // Still, we should make this case safe for when we have a non-zero 'exploderScale' -
+  vec3 offset = pos - Mesh.exploderCentre;
+  float offsetDistSquared = dot(offset, offset);
+  vec3 safeExplodeDir = offset * inversesqrt(max(offsetDistSquared, FLT_EPSILON));
+
+  float displacement = nonLinearVtxExplodeScale * ((float((vtxID >> 1) & 0xf) / 15.0f) * 1.5f - 0.5f);
+  pos += (safeExplodeDir * displacement);
+
+  if(Mesh.exploderScale > 0.0f)
```

Comment:
I think it works out fine either way, but is there a reason this whole function isn't conditional on `Mesh.exploderScale > 0.0f` ?